src/diffusers/models/unet_2d_condition.py

```diff
@@ -163,6 +163,7 @@ class conditioning with `class_embed_type` equal to `None`.
     """
 
     _supports_gradient_checkpointing = True
+    _no_split_modules = ["Block"]
```

Comment:
I'm a little confused as to what should go here. Seems like all the blocks having "Block" in their names (such as `DownBlock2D`, `UNetMidBlock2DCrossAttn`, etc.) should go here as they have residual connections inside of them (recommendation comes from [here](https://huggingface.co/docs/accelerate/concept_guides/big_model_inference#nosplitmoduleclasses)).

Reply:
Yeah, that's right. We need to add all potential blocks name that have residual connections inside of them. What is tricky with diffusers models it that it allows different types of transformers block (e.g. `DownBlock2D`, `CrossAttnDownBlock2D`) which is not the case in transformers. This makes the testing harder.

Reply:
That indeed sounds tricky. 

So, let's think this through.

If you take at the config.json of the UNet of SDXL [here](https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/blob/main/unet/config.json), you would notice the distinct block types there. Now these blocks are composed of lower-level blocks that have residual connections. Would adding these higher-level blocks cut it for us?

Reply:
We can definitely add these higher level-blocks instead of the lower-level blocks if they are not occupying too much memory space. I think that if we can have blocks of  < ~1GB, it should be good enough.

Reply:
> I think that if we can have blocks of < ~1GB, it should be good enough.

Could you elaborate this a bit? Additionally, it would be helpful if you maybe locally test this PR and let me know your findings, that would be great!

Reply:
I probably got confused by the term higher level-block. Can you explain what do you mean by that ? I was saying that it is okay to not split high level modules if they are small enough. 
For example, in this example where the submodules have residual connections, the best case would be to set `_no_split_modules = [SubModule 1, SubModule 2, SubModule 3].` But we can decide to set `_no_split_modules = [Module 1]` if Module 1 is not too big. 
(Module 1)
----(SubModule 1)
----(SubModule 2)
----(SubModule 3)

Reply:
Asked a question over Slack.